src/main/java/ru/hogwarts/school/controller/FacultyController.java

```diff
@@ -46,10 +46,23 @@ public ResponseEntity<Void> deleteFaculty(@RequestParam Long id) {
     }
 
     @GetMapping
-    public ResponseEntity<Collection<Faculty>> getStudentsCertainColor(@RequestParam String color) {
+    public ResponseEntity<Collection<Faculty>> getStudentsByNameOrColor(@RequestParam(required = false) String color,
+                                                                        @RequestParam(required = false) String name) {
         if (color != null && !color.isBlank()) {
-            return ResponseEntity.ok(facultyService.findFacultyByColor(color));
+            return ResponseEntity.ok(facultyService.findFacultyByNameOrColor(name, color));
+        }
+        if (name != null && !name.isBlank()) {
+            return ResponseEntity.ok(facultyService.findFacultyByNameOrColor(name, color));
         }
         return ResponseEntity.ok(Collections.emptyList());
     }
+
+    @GetMapping("/getStudents")
+    public ResponseEntity<Collection<Student>> getFacultyStudents(@RequestParam Long id) {
```

Comment:
Неправильно запрос сделан с точки зрения правил REST. Во-первых, надо обратиться по id к конкретному ресурсу. Во-вторых, надо получить его свойство

```suggestion
    @GetMapping("/{id}/students")
    public ResponseEntity<Collection<Student>> getFacultyStudents(@PathVariable Long id) {
```